unittests/Interpreter/InterpreterTests.cpp

```diff
@@ -0,0 +1,442 @@
+    }
+    break;
+    case Inst::Ult:
+    {
+      res = KnownBits(1);
+      if (xc.ult(yc))
+	res.One.setBit(0);
+      else
+	res.Zero.setBit(0);
+    }
+    break;
+    case Inst::Slt:
+    {
+      res = KnownBits(1);
+      if (xc.slt(yc))
+	res.One.setBit(0);
+      else
+	res.Zero.setBit(0);
+    }
+    break;
+    case Inst::Ule:
+    {
+      res = KnownBits(1);
+      if (xc.ule(yc))
+	res.One.setBit(0);
+      else
+	res.Zero.setBit(0);
+    }
+    break;
+    case Inst::Sle:
+    {
+      res = KnownBits(1);
+      if (xc.sle(yc))
+	res.One.setBit(0);
+      else
+	res.Zero.setBit(0);
+    }
+    break;
+    default:
+      break;
+    }
+    return res;
+  }
+
+  bool testFn(Inst::Kind pred) {
+    llvm::KnownBits x(WIDTH);
+    do {
+      llvm::KnownBits y(WIDTH);
+      do {
+	KnownBits Res1;
+
+	switch(pred) {
+	case Inst::AddNUW:
+	case Inst::AddNW:
+	case Inst::Add:
+	  Res1 = BinaryTransferFunctionsKB::add(x, y);
+	  break;
+	case Inst::AddNSW:
+	  Res1 = BinaryTransferFunctionsKB::addnsw(x, y);
+	  break;
+	case Inst::SubNUW:
+	case Inst::SubNW:
+	case Inst::Sub:
+	  Res1 = BinaryTransferFunctionsKB::sub(x, y);
+	  break;
+	case Inst::SubNSW:
+	  Res1 = BinaryTransferFunctionsKB::subnsw(x, y);
+	  break;
+	case Inst::Mul:
+	  Res1 = BinaryTransferFunctionsKB::mul(x, y);
+	  break;
+	case Inst::UDiv:
+	  Res1 = BinaryTransferFunctionsKB::udiv(x, y);
+	  break;
+	case Inst::URem:
+	  Res1 = BinaryTransferFunctionsKB::urem(x, y);
+	  break;
+	case Inst::And:
+	  Res1 = BinaryTransferFunctionsKB::and_(x, y);
+	  break;
+	case Inst::Or:
+	  Res1 = BinaryTransferFunctionsKB::or_(x, y);
+	  break;
+	case Inst::Xor:
+	  Res1 = BinaryTransferFunctionsKB::xor_(x, y);
+	  break;
+	case Inst::Shl:
+	  Res1 = BinaryTransferFunctionsKB::shl(x, y);
+	  break;
+	case Inst::LShr:
+	  Res1 = BinaryTransferFunctionsKB::lshr(x, y);
+	  break;
+	case Inst::AShr:
+	  Res1 = BinaryTransferFunctionsKB::ashr(x, y);
+	  break;
+	case Inst::Eq:
+	  Res1 = BinaryTransferFunctionsKB::eq(x, y);
+	  break;
+	case Inst::Ne:
+	  Res1 = BinaryTransferFunctionsKB::ne(x, y);
+	  break;
+	case Inst::Ult:
+	  Res1 = BinaryTransferFunctionsKB::ult(x, y);
+	  break;
+	case Inst::Slt:
+	  Res1 = BinaryTransferFunctionsKB::slt(x, y);
+	  break;
+	case Inst::Ule:
+	  Res1 = BinaryTransferFunctionsKB::ule(x, y);
+	  break;
+	case Inst::Sle:
+	  Res1 = BinaryTransferFunctionsKB::sle(x, y);
+	  break;
+	default:
+	  break;
+	}
+
+	KnownBits Res2 = bruteForce(x, y, pred);
+	if (Res1.getBitWidth() != Res2.getBitWidth()) {
+	  llvm::errs() << "Expected and Given have unequal bitwidths - Expected: "
+		       << Res2.getBitWidth() << ", Given: " << Res1.getBitWidth() << '\n';
+	  return false;
+	}
+	if (Res1.hasConflict() || Res2.hasConflict()) {
+	  llvm::errs() << "Expected or Given result has a conflict\n";
+	  return false;
+	}
+
+	for (unsigned i = 0; i < Res1.getBitWidth(); i++) {
+	  if ((Res1.Zero[i] and Res2.One[i]) || (Res1.One[i] and Res2.Zero[i])) {
```

Comment:
let's avoid the alternative operators like "and" unless there's a compelling reason to use